webapp/courses/templates/courses/exercise-evaluation-messages.html

```diff
@@ -1,9 +1,68 @@
 {% load i18n %}
-{% for message in messages %}
-  <ul>
-    <li class="evaluation-msg-title">{{ message.0 }}</li>
-    {% for msg_detail in message.1 %}
-      <li>{{ msg_detail|safe }}</li>
+{% for test in log %}
+    <div class="evaluation-msg-title">
+    {{ test.title|safe }}
+    </div>  
+    {% for run in test.runs %}
+        {% if forloop.first %}
+        {% if run.output %}
+        <div class="evaluation-run evaluation-run-open" onclick="collapse_eval_debug(this)">
```

Comment:
Both the template language tags and HTML tags should follow typical indentation rules (nesting something -> indent).